xbdd-web/src/modules/summary/SummaryContainer.jsx

```diff
@@ -0,0 +1,109 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { Grid, Card, Typography } from "@material-ui/core";
+import { withStyles } from "@material-ui/core/styles";
+import ProductListContainer from "./productList/ProductListContainer";
+import SummaryStyles from "./styles/SummaryStyles";
+import ProductSummary from "../../models/ProductSummary";
+import { getSummaryOfReports, setProductFavouriteOn, setProductFavouriteOff } from "../../lib/rest/Rest";
+
+class SummaryContainer extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { productList: null, favouriteList: null };
+
+    this.handleFavouriteChange = this.handleFavouriteChange.bind(this);
+  }
+
+  componentDidMount() {
+    getSummaryOfReports().then(summaryData => {
+      const productList = new ProductSummary(summaryData).productList;
+      this.setState({
+        productList: this.processList(productList, false),
+        favouriteList: this.processList(productList, true),
+      });
+    });
+  }
+
+  processList(productList, isFavouriteList) {
+    const list = [];
+    productList.forEach(element => {
+      var flag = true;
+      list.forEach(product => {
+        if (product.name === element.name) {
+          flag = false;
+        }
+      });
+      if (flag) {
+        if (!isFavouriteList) {
```

Comment:
This part of the function could also be cleaned up. We could have a single variable like `const shouldBeAdded = isFavouriteList ? !hasBeenAdded && element.favourite : !hasBeenAdded;` then a single guard on pushing the element to the list.